language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -720,6 +738,110 @@ describe("interfaces", () => {
       },
     ]);
   });
+
+  test("sync query without predicate", async () => {
+    const { aliceLedger, expectedView, contract } =
+      await aliceLedgerPayloadContract();
+    const acs = await aliceLedger.query(Token);
+    const expectedAc: typeof acs extends (infer ce)[]
+      ? Omit<ce, "key">
+      : never = {
+      templateId: Token.templateId, // NB: the interface ID
+      contractId: Asset.toInterface(Token, contract.contractId),
+      signatories: [ALICE_PARTY],
+      observers: [],
+      agreementText: "",
+      payload: expectedView,
+    };
+    expect(acs).toContainEqual(expectedAc);
+  });
+
+  test("sync query with predicate", async () => {
+    const { aliceLedger, expectedView, contract } =
+      await aliceLedgerPayloadContract();
+    function isCt(ev: CreateEvent<Token>) {
+      return ev.contractId === Asset.toInterface(Token, contract.contractId);
+    }
+    // check that Query type accepts removing the brand
+    const succeedingQuery: Query<Token> = {
+      tokenOwner: ALICE_PARTY,
+    };
+    const failingQuery = { tokenOwner: BOB_PARTY };
+
+    const foundCt = (await aliceLedger.query(Token, succeedingQuery)).filter(
+      isCt,
+    );
+    const noCt = (await aliceLedger.query(Token, failingQuery)).filter(isCt);
+    const expectedMatchedContract = {
+      payload: expectedView,
+      templateId: Token.templateId,
+    };
+    expect(foundCt).toMatchObject([expectedMatchedContract]);
+    expect(noCt).toEqual([]);
+  });
+
+  test("fetch", async () => {
+    const { aliceLedger, expectedView, contract } =
+      await aliceLedgerPayloadContract();
+    const fetched = await aliceLedger.fetch(
+      Token,
+      Asset.toInterface(Token, contract.contractId),
+    );
+    expect(fetched).toMatchObject({
+      contractId: contract.contractId,
+      templateId: Token.templateId,
+      payload: expectedView,
+    });
+  });
+
+  // TODO #14920 enable
+  test.skip("WS query", async () => {
```

Comment:
This test passes against 241415b318320ea11779f2c19658205bb737204e from #14919; just have to enable it.